processor/compile/compatibility.ts

```diff
@@ -8,6 +11,22 @@ type CompatNodes =
   | { type: NodeTypes.reusableContent; tag: string }
   | Html;
 
+/*
+ * Converts a (remark < v9) html node to a JSX string.
+ *
+ * First we replace html comments with the JSX equivalent. Then, we parse that
+ * as html, and serialize it back as xml!
+ *
+ */
+const html = (node: Html) => {
+  const string = node.value.replaceAll(/<!--(.*)-->/gms, '{/*$1*/}');
+  const hast = fromHtml(string);
+  const xast = toXast(hast);
+  const xml = toXml(xast, { closeEmptyElements: true });
```

Comment:
This is a smart approach; I was worried [when you said manually](https://readmeio.slack.com/archives/C04R9PADCS3/p1717440549009189?thread_ts=1717434083.400309&cid=C04R9PADCS3) we'd actually be manually inserting the `/>` ourselves!

Reply:
In that context, I was trying to refer to when they html is malformed, like it doesn't have a closing tag:
```
<p>
...
```

But, yea I think this is much better than a regex too.